packages/block-library/src/media-text/edit.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	BlockControls,
+	InnerBlocks,
+	InspectorControls,
+	PanelColorSettings,
+	withColors,
+} from '@wordpress/editor';
+import { Component, Fragment } from '@wordpress/element';
+import { Toolbar } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import MediaContainer from './media-container';
+
+/**
+ * Constants
+ */
+const ALLOWED_BLOCKS = [ 'core/button', 'core/paragraph', 'core/heading', 'core/list' ];
+const TEMPLATE = [
+	[ 'core/paragraph', { fontSize: 'large', placeholder: 'Content…' } ],
+];
+
+class MediaTextEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onSelectMedia = this.onSelectMedia.bind( this );
+		this.onWidthChange = this.onWidthChange.bind( this );
+		this.commitWidthChange = this.commitWidthChange.bind( this );
+		this.state = {
+			mediaWidth: null,
+		};
+	}
+
+	onSelectMedia( media ) {
+		const { setAttributes } = this.props;
+
+		let mediaType;
+		// for media selections originated from a file upload.
+		if ( media.media_type ) {
+			if ( media.media_type === 'image' ) {
+				mediaType = 'image';
+			} else {
+				// only images and videos are accepted so if the media_type is not an image we can assume it is a video.
+				// video contain the media type of 'file' in the object returned from the rest api.
+				mediaType = 'video';
+			}
+		} else { // for media selections originated from existing files in the media library.
+			mediaType = media.type;
+		}
+
+		setAttributes( {
+			mediaAlt: media.alt,
+			mediaId: media.id,
+			mediaType,
+			mediaUrl: media.url,
+		} );
+	}
+
+	onWidthChange( width ) {
+		this.setState( {
+			mediaWidth: width,
+		} );
+	}
+
+	commitWidthChange( width ) {
+		const { setAttributes } = this.props;
+
+		setAttributes( {
+			mediaWidth: width,
+		} );
+		this.setState( {
+			mediaWidth: null,
+		} );
+	}
+
+	renderMediaArea() {
+		const { attributes } = this.props;
+		const { mediaAlt, mediaId, mediaPosition, mediaType, mediaUrl, mediaWidth } = attributes;
+
+		return (
+			<MediaContainer
+				className="block-library-media-text__media-container"
+				onSelectMedia={ this.onSelectMedia }
+				onWidthChange={ this.onWidthChange }
+				commitWidthChange={ this.commitWidthChange }
+				{ ...{ mediaAlt, mediaId, mediaType, mediaUrl, mediaPosition, mediaWidth } }
+			/>
+		);
+	}
+
+	render() {
+		const { attributes, backgroundColor, setAttributes, setBackgroundColor } = this.props;
+		const { mediaPosition, mediaWidth } = attributes;
+		const temporaryMediaWidth = this.state.mediaWidth;
+		const className = classnames( 'wp-block-media-text', {
```

Comment:
I think the best practice is to use the "className" prop instead of hardcoding this 'wp-block-media-text'